extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryUtil.java

```diff
@@ -49,20 +54,27 @@ public static Map<String, String> convertKeyValueListToMap(List<String> headers)
     }
 
     /**
-     * Sets MDC data by using the current span from the context.
+     * Sets MDC data by using the current span from the otelContext.
      * <p>
      * This method is in the hot path and was optimized to not use getSpanData()
      *
-     * @param context opentelemetry context
-     * @param vertxContext vertx context
+     * @param otelContext opentelemetry otelContext
+     * @param vertxContext vertx otelContext
      */
-    public static void setMDCData(Context context, io.vertx.core.Context vertxContext) {
-        if (context == null) {
+    public static void setMDCData(Context otelContext, io.vertx.core.Context vertxContext) {
+        if (otelContext == null) {
             return;
         }
 
-        Span span = Span.fromContextOrNull(context);
+        Span span = Span.fromContextOrNull(otelContext);
         if (span != null) {
+            logger.debugv("Setting span in MDC: {0} with otel data in vertx context: {1}",
+                    getSpanData(otelContext),
+                    getSpanData(QuarkusContextStorage.getOtelContext(vertxContext)));
```

Comment:
Let's put this in an `if (logger.isDebugEnabled())` as it performs operations that will not be used most of the time.